File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n = nums.size();
        vector<int> ans;
        unordered_map<int,int> mp;
        for(int i=0;i<n;i++){
            mp[nums[i]]=i;
        }

        for(int i=0;i<n;i++){
            int diff = target-nums[i];

            if(mp.find(diff)!=mp.end() && mp[diff]!=i){
                ans.push_back(i);
                ans.push_back(mp[diff]);
                break;
            }
        }
        return ans;
    }
};

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        
        HashMap<Integer, Integer> m1 = new HashMap<>();
        int []ans = new int[2];
        for(int i=0;i<nums.length; i++){
            m1.put(nums[i], i);
        }

        for(int i=0;i<nums.length; i++){
            int diff = target - nums[i];

            if(m1.containsKey(diff) && m1.get(diff)!=i){
                ans[0] = i;
                ans[1] = m1.get(diff);
                break;
            }
        }
        return ans;
    }
}